File: backend/swim-admin/src/main/java/com/smartlight/entity/Streetlight.java
package com.smartlight.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "streetlight")
public class Streetlight {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 50)
    private String code;

    @Column(name = "device_uid", length = 100)
    private String deviceUid;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(name = "area_id", nullable = false)
    private Long areaId;

    @Column(name = "cabinet_id")
    private Long cabinetId;

    @Column(length = 500)
    private String address;

    @Column(precision = 10, scale = 7)
    private BigDecimal longitude;

    @Column(precision = 10, scale = 7)
    private BigDecimal latitude;

    @Column(name = "lamp_type", length = 50)
    private String lampType;

    @Column(name = "hardware_model", length = 100)
    private String hardwareModel;

    @Column(name = "electrical_params", length = 500)
    private String electricalParams;

    @Column(name = "protection_rating", length = 20)
    private String protectionRating;

    private Integer power;

    @Column(precision = 5, scale = 2)
    private BigDecimal height;

    private Integer brightness = 100;

    @Column(name = "online_status", nullable = false)
    private Integer onlineStatus = 1;

    @Column(name = "light_status", nullable = false)
    private Integer lightStatus = 0;

    @Column(name = "device_status", nullable = false)
    private Integer deviceStatus = 1;

    @Column(name = "install_date")
    private LocalDate installDate;

    @Column(name = "last_maintain_date")
    private LocalDate lastMaintainDate;

    @Column(precision = 6, scale = 2)
    private BigDecimal voltage;

    @Column(name = "current_val", precision = 6, scale = 3)
    private BigDecimal currentVal;

    @Column(precision = 5, scale = 2)
    private BigDecimal temperature;

    @Column(name = "running_hours")
    private Integer runningHours = 0;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "area_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Area area;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cabinet_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "area"})
    private Cabinet cabinet;

    @PrePersist
    protected void onCreate() { createdAt = LocalDateTime.now(); updatedAt = LocalDateTime.now(); }

    @PreUpdate
    protected void onUpdate() { updatedAt = LocalDateTime.now(); }
}

File: backend-csharp/SmartStreetlight.Api/Models/Entities/Streetlight.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace SmartStreetlight.Api.Models.Entities;

[Table("streetlight")]
public class Streetlight
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    [Required, MaxLength(50)]
    public string Code { get; set; } = "";

    [Column("device_uid"), MaxLength(100)]
    public string? DeviceUid { get; set; }

    [Required, MaxLength(100)]
    public string Name { get; set; } = "";

    [Column("area_id")]
    public long AreaId { get; set; }

    [Column("cabinet_id")]
    public long? CabinetId { get; set; }

    [MaxLength(500)]
    public string? Address { get; set; }

    [Column(TypeName = "decimal(10,7)")]
    public decimal? Longitude { get; set; }

    [Column(TypeName = "decimal(10,7)")]
    public decimal? Latitude { get; set; }

    [Column("lamp_type"), MaxLength(50)]
    public string? LampType { get; set; }

    [Column("hardware_model"), MaxLength(100)]
    public string? HardwareModel { get; set; }

    [Column("electrical_params"), MaxLength(500)]
    public string? ElectricalParams { get; set; }

    [Column("protection_rating"), MaxLength(20)]
    public string? ProtectionRating { get; set; }

    public int? Power { get; set; }

    [Column(TypeName = "decimal(5,2)")]
    public decimal? Height { get; set; }

    public int Brightness { get; set; } = 100;

    [Column("online_status")]
    public int OnlineStatus { get; set; } = 1;

    [Column("light_status")]
    public int LightStatus { get; set; } = 0;

    [Column("device_status")]
    public int DeviceStatus { get; set; } = 1;

    [Column("install_date")]
    public DateOnly? InstallDate { get; set; }

    [Column("last_maintain_date")]
    public DateOnly? LastMaintainDate { get; set; }

    [Column(TypeName = "decimal(6,2)")]
    public decimal? Voltage { get; set; }

    [Column("current_val", TypeName = "decimal(6,3)")]
    public decimal? CurrentVal { get; set; }

    [Column(TypeName = "decimal(5,2)")]
    public decimal? Temperature { get; set; }

    [Column("running_hours")]
    public int RunningHours { get; set; } = 0;

    [Column("created_at")]
    [JsonIgnore]
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    [Column("updated_at")]
    [JsonIgnore]
    public DateTime UpdatedAt { get; set; } = DateTime.Now;

    // Navigation
    [ForeignKey("AreaId")]
    public Area? Area { get; set; }

    [ForeignKey("CabinetId")]
    public Cabinet? Cabinet { get; set; }
}
